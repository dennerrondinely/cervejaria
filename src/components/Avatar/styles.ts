import styled from 'styled-components';

export const Wrapper = styled.div`
  height: 7.8rem;
  width: 7.8rem;
  background: #ffc366;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 3.9rem;
`;

export const Letter = styled.span`
  font-family: 'Roboto';
  font-style: normal;
  font-weight: 700;
  font-size: 42px;
  line-height: 28px;
  text-align: center;
  letter-spacing: 0.35px;
  text-transform: uppercase;

  color: #8a7251;
`;
