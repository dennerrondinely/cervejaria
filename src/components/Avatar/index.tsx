import * as S from './styles';

export type AvatarProps = {
  name?: string;
};

const Avatar = ({ name }: AvatarProps) => {
  const splitName = name ? name.split('') : ['-'];
  const [letter] = splitName.filter((l) => isNaN(Number(l)));
  return (
    <S.Wrapper>
      <S.Letter>{letter}</S.Letter>
    </S.Wrapper>
  );
};

export default Avatar;
