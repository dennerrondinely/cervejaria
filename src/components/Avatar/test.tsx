import Avatar from './index';
import { render, screen } from 'utils/tests/helpers';

describe('<Avatar />', () => {
  it('should render the Avatar', () => {
    const { container } = render(<Avatar name="10 Barrel Brewing Co" />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should render the Avatar withou name', () => {
    render(<Avatar />);
    const defaulyValue = screen.getByText(/-/i);
    expect(defaulyValue).toBeInTheDocument();
  });
});
