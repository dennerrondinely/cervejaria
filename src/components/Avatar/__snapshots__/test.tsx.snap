// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render the Avatar 1`] = `
.c0 {
  height: 7.8rem;
  width: 7.8rem;
  background: #ffc366;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 3.9rem;
}

.c1 {
  font-family: 'Roboto';
  font-style: normal;
  font-weight: 700;
  font-size: 42px;
  line-height: 28px;
  text-align: center;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  text-transform: uppercase;
  color: #8a7251;
}

<div
  class="c0"
>
  <span
    class="c1"
  >
    B
  </span>
</div>
`;
